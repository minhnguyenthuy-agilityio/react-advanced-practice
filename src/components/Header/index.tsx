import { Avatar, Box, Flex, Icon, Text } from "@chakra-ui/react";

import { SearchBox } from "@/components";

import { USER } from "@/mocks";

import { ArrowDownIcon } from "@/assets/icons";

export const Header = () => {
  return (
    <Flex
      as="header"
      borderBottom="1px solid"
      borderColor="primary.400"
      px="55px"
      py="25px"
      justifyContent="space-between"
      alignItems="center"
    >
      <SearchBox />
      <Flex alignItems="center">
        <Box mr="20px" textAlign="right">
          <Text color="primary.700">{USER.name}</Text>
          <Text size="sm">{USER.country}</Text>
        </Box>
        <Avatar mr="10px" w="38px" h="38px" src={USER.avatar} />
        <Icon as={ArrowDownIcon} />
      </Flex>
    </Flex>
  );
};
