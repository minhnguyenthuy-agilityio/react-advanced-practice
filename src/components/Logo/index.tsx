import { Link as ChakraLink, Heading, Icon, Text } from "@chakra-ui/react";

import { Link as ReactRouterLink } from "react-router-dom";

import { LogoIcon } from "@/assets/icons";

import { ROUTES } from "@/constants";

export const Logo = () => (
  <Heading
    as="h1"
    py="35px"
    pl="20px"
    alignItems="center"
    borderBottom="solid 1px "
    borderColor="primary.400"
  >
    <ChakraLink
      display="flex"
      as={ReactRouterLink}
      to={ROUTES.TASKS_PROJECT_DEFAULT}
      _hover={{
        textDecoration: "none",
      }}
    >
      <Icon mr="8px" w="24px" h="24px" as={LogoIcon} />
      <Text color="primary.700" size="xm" fontWeight="semibold">
        Project M.
      </Text>
    </ChakraLink>
  </Heading>
);
