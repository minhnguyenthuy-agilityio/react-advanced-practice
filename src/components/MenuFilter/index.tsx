import { FC, memo } from "react";

import {
  Button,
  Icon,
  Menu,
  MenuButton,
  MenuList,
  MenuItem,
} from "@chakra-ui/react";

import { ArrowDownFilterIcon } from "@/assets/icons";

interface MenuFilterProps {
  buttonName: string;
  options: string[];
  icon?: FC<React.SVGProps<SVGSVGElement>>;
}

export const MenuFilter = memo(
  ({ buttonName, options, icon }: MenuFilterProps) => {
    return (
      <Menu>
        <MenuButton
          as={Button}
          variant="outline"
          leftIcon={icon && <Icon data-testid="icon-button-menu" as={icon} />}
          rightIcon={<Icon as={ArrowDownFilterIcon} />}
          w="122px"
          maxH="40px"
          fontWeight="medium"
        >
          {buttonName}
        </MenuButton>
        <MenuList>
          {options?.map((option, index) => (
            <MenuItem
              data-testid="menu-item"
              key={`${option}-${index}`}
              _hover={{
                bgColor: "secondary.200",
              }}
            >
              {option}
            </MenuItem>
          ))}
        </MenuList>
      </Menu>
    );
  },
);
