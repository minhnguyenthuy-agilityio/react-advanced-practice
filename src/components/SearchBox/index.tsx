import { ChangeEvent, KeyboardEvent, useState } from "react";

import { Icon, Input, InputGroup, InputLeftElement } from "@chakra-ui/react";

import { SearchIcon } from "@/assets/icons";

import { useTaskListStore } from "@/hooks";

export const SearchBox = () => {
  const [keyValue, setKeyValue] = useState("");
  const { setKeyword } = useTaskListStore();

  const handleOnChange = (
    event: ChangeEvent<
      HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement
    >,
  ) => {
    setKeyValue(event.target.value);
  };

  const handleKeyDown = async (event: KeyboardEvent<HTMLInputElement>) => {
    if (event.key === "Enter") {
      setKeyword(keyValue.trim());
    }
  };

  return (
    <InputGroup bg="primary.200" borderRadius="6px" maxW="417px">
      <InputLeftElement left="12px" top="10%" fontSize="sm">
        <Icon as={SearchIcon} />
      </InputLeftElement>
      <Input
        pl="55px"
        h="full"
        placeholder="Search for anything..."
        py="13px"
        onChange={handleOnChange}
        onKeyDown={handleKeyDown}
      />
    </InputGroup>
  );
};
