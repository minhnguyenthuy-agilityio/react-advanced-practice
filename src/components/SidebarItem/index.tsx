import { Link as ChakraLink, Icon, Text } from "@chakra-ui/react";

import { DotIcon } from "@/assets/icons";

import { ROUTES } from "@/constants";

import { Link as ReactRouterLink } from "react-router-dom";

interface SidebarItemPropType {
  iconColor: string;
  title: string;
  projectId: string;
  isActive?: boolean;
}

export const SidebarItem = ({
  iconColor,
  title,
  projectId,
  isActive,
}: SidebarItemPropType) => {
  return (
    <ChakraLink
      as={ReactRouterLink}
      to={`${ROUTES.TASKS_PROJECT_DEFAULT}${projectId}`}
      display="flex"
      w="full"
      pl="14px"
      py="12px"
      borderRadius="6px"
      alignItems="center"
      background={isActive ? "secondary.200" : "transparent"}
      _hover={{
        textDecoration: "none",
        bgColor: "secondary.200",
      }}
    >
      <Icon mr="16px" as={DotIcon} color={iconColor} />
      <Text
        textOverflow="ellipsis"
        whiteSpace="nowrap"
        color={isActive ? "primary.700" : "primary.500"}
        fontWeight={isActive ? "semibold" : "medium"}
        overflowX="hidden"
      >
        {title}
      </Text>
    </ChakraLink>
  );
};
