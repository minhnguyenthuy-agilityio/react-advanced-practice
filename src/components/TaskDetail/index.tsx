import {
  Modal,
  ModalOverlay,
  ModalContent,
  ModalHeader,
  ModalCloseButton,
  ModalBody,
  Badge,
  Heading,
  Flex,
  AvatarGroup,
  Avatar,
  Icon,
  Text,
  Image,
  Center,
} from "@chakra-ui/react";

import { PRIORITY_MAPPING, Status } from "@/constants";

import { Task } from "@/interfaces";

import { FolderIcon, MessageIcon } from "@/assets/icons";

import { Gallery } from "@/assets/images";

interface TaskDetailProps extends Omit<Task, "id" | "projectId" | "createdAt"> {
  isOpenDetail: boolean;
  onCloseDetail: () => void;
}

export const TaskDetail = ({
  isOpenDetail,
  onCloseDetail,
  priority,
  name,
  description,
  image,
  assignees,
  comments,
  files,
  status,
}: TaskDetailProps) => {
  const { color, bgColor, label } = PRIORITY_MAPPING[priority];

  return (
    <Modal isOpen={isOpenDetail} onClose={onCloseDetail}>
      <ModalOverlay />
      <ModalContent maxW="848px" h="80%" px="70px" py="60px" borderRadius="30">
        <ModalHeader p="0px">
          <Badge
            color={status === Status.Done ? "success.400" : color}
            bgColor={status === Status.Done ? "success.300" : bgColor}
          >
            {status === Status.Done ? "Completed" : label}
          </Badge>

          <Heading size="lg" my="20px" as="h3">
            {name}
          </Heading>
        </ModalHeader>
        <ModalCloseButton top="30px" right="20px" />
        <ModalBody p="0px" overflowY="auto">
          {image ? (
            <Image
              mb="10px"
              w="full"
              h="233px"
              borderRadius="6px"
              alt="background task"
              objectFit="cover"
              src={image}
            />
          ) : (
            <Center
              mb="10px"
              bg="primary.300"
              borderRadius="10"
              py="50px"
              w="full"
              h="233px"
            >
              <Image
                borderRadius="6px"
                h="40px"
                w="40px"
                alt="background task"
                src={Gallery}
              />
            </Center>
          )}

          <Flex my="20px" justifyContent="space-between" alignItems="center">
            <AvatarGroup spacing="-4px" size="xs" max={5}>
              {assignees?.map(({ avatar, name }, index) => (
                <Avatar name="assignee" key={`${name}-${index}`} src={avatar} />
              ))}
            </AvatarGroup>
            <Flex gap={2}>
              <Flex>
                <Icon as={MessageIcon} />
                <Text ml={1} size="xs">
                  {`${comments} ${comments <= 1 ? "comment" : "comments"}`}
                </Text>
              </Flex>
              <Flex>
                <Icon as={FolderIcon} />
                <Text ml={1} size="xs">
                  {`${files} ${files <= 1 ? "file" : "files"}`}
                </Text>
              </Flex>
            </Flex>
          </Flex>

          <Text color={description ? "secondary.400" : "primary.500"}>
            {description ? description : " No description provided"}
          </Text>
        </ModalBody>
      </ModalContent>
    </Modal>
  );
};
