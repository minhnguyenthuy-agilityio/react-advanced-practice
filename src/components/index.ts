export * from "./Header";
export * from "./Sidebar";
export * from "./SearchBox";
export * from "./Logo";
export * from "./SidebarItem";
export * from "./TaskCard";
export * from "./TaskDataEntryForm";
export * from "./MenuFilter";
export * from "./TaskDetail";
export * from "./TaskList";
