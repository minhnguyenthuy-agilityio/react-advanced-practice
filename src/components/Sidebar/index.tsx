import { Box, Flex, Icon, Text, VStack } from "@chakra-ui/react";

import { useParams } from "react-router-dom";

import { Logo, SidebarItem } from "@/components";

import { PROJECTS } from "@/mocks";

import { TasksIcon } from "@/assets/icons";

export const SideBar = () => {
  const { projectId } = useParams();
  const project = projectId ? projectId : "";

  return (
    <Box as="aside" h="full" borderRight="1px solid" borderColor="primary.400">
      <Logo />
      <Box pr="10px" pl="2px">
        <Flex
          alignItems="center"
          pl="16px"
          py="14px"
          borderBottom="solid 1px"
          borderColor="primary.400"
        >
          <Icon mr="14px" as={TasksIcon} />
          <Text>Tasks</Text>
        </Flex>

        <VStack as="nav" spacing="2.5" alignItems="start">
          <Text
            pl="3"
            pt="7"
            pb="2"
            textTransform="uppercase"
            size="xs"
            fontWeight="700"
          >
            my projects
          </Text>
          {PROJECTS?.map(({ name, iconColor, projectId }, index) => (
            <SidebarItem
              key={`${name}-${index}`}
              iconColor={iconColor}
              title={name}
              projectId={projectId}
              isActive={project === projectId}
            />
          ))}
        </VStack>
      </Box>
    </Box>
  );
};
