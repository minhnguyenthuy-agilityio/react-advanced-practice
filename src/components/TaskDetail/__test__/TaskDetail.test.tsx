import { render } from "@/helpers";

import { TaskDetail } from "@/components";

import { TASK } from "@/mocks";

import { Gallery } from "@/assets/images";

import { Priority } from "@/constants";

describe("TaskDetailItem component", () => {
  const taskInformation = {
    isOpenDetail: true,
    onCloseDetail: jest.fn(),
    name: "Sample Task",
    priority: Priority.High,
    description: "Sample description for the task",
    image:
      "https://res.cloudinary.com/de2x8dwvf/image/upload/v1695227531/test_y38vvw.webp",
    assignees: TASK.assignees,
    files: TASK.files,
    comments: TASK.comments,
    status: 0,
  };

  it("matchs snapshoot", () => {
    const element = render(<TaskDetail {...taskInformation} />);
    expect(element).toMatchSnapshot();
  });

  it("renders TaskDetail with correct details", () => {
    const { getByText } = render(<TaskDetail {...taskInformation} />);

    expect(getByText(taskInformation.name)).toBeInTheDocument();
    expect(getByText(taskInformation.description)).toBeInTheDocument();

    expect(getByText("20 comments")).toBeInTheDocument();
    expect(getByText("15 files")).toBeInTheDocument();
  });

  it("renders TaskCard with default image when image is not provided", () => {
    const taskInformationNoImage = {
      ...taskInformation,
      image: "",
    };
    const { getByAltText } = render(<TaskDetail {...taskInformationNoImage} />);

    const defaultImageElement = getByAltText("background task");
    expect(defaultImageElement).toBeInTheDocument();
    expect(defaultImageElement).toHaveAttribute("src", Gallery);
  });
});
