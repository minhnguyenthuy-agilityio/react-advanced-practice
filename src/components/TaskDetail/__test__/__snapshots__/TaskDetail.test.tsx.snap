// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskDetailItem component matchs snapshoot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      aria-hidden="true"
      data-aria-hidden="true"
    >
      <span
        hidden=""
        id="__chakra_env"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <div
        aria-hidden="true"
        class="chakra-modal__overlay css-hdd9l7"
        data-aria-hidden="true"
        style="opacity: 0;"
      />
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
      <div
        data-focus-lock-disabled="false"
      >
        <div
          class="chakra-modal__content-container css-1u2cvaz"
          tabindex="-1"
        >
          <section
            aria-describedby="chakra-modal--body-:r0:"
            aria-labelledby="chakra-modal--header-:r0:"
            aria-modal="true"
            class="chakra-modal__content css-wkqcrs"
            id="chakra-modal-:r0:"
            role="dialog"
            style="opacity: 0; transform: scale(0.95) translateZ(0);"
            tabindex="-1"
          >
            <header
              class="chakra-modal__header css-f8atus"
              id="chakra-modal--header-:r0:"
            >
              <span
                class="chakra-badge css-1uudyz1"
              >
                High
              </span>
              <h3
                class="chakra-heading css-4hctp9"
              >
                Sample Task
              </h3>
            </header>
            <button
              aria-label="Close"
              class="chakra-modal__close-btn css-1dl8qrb"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <div
              class="chakra-modal__body css-14hgxoh"
              id="chakra-modal--body-:r0:"
            >
              <img
                alt="background task"
                class="chakra-image css-1k2e4o5"
                src="https://res.cloudinary.com/de2x8dwvf/image/upload/v1695227531/test_y38vvw.webp"
              />
              <div
                class="css-vnjiam"
              >
                <div
                  class="chakra-avatar__group css-1rs5v3u"
                  role="group"
                >
                  <span
                    class="chakra-avatar css-106sop3"
                  >
                    <svg
                      aria-label=" avatar"
                      class="chakra-avatar__svg css-16ite8i"
                      role="img"
                      viewBox="0 0 128 128"
                    >
                      <path
                        d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                        fill="currentColor"
                      />
                      <path
                        d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                  <span
                    class="chakra-avatar css-rawe7k"
                  >
                    <svg
                      aria-label=" avatar"
                      class="chakra-avatar__svg css-16ite8i"
                      role="img"
                      viewBox="0 0 128 128"
                    >
                      <path
                        d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                        fill="currentColor"
                      />
                      <path
                        d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                  <span
                    class="chakra-avatar css-rawe7k"
                  >
                    <svg
                      aria-label=" avatar"
                      class="chakra-avatar__svg css-16ite8i"
                      role="img"
                      viewBox="0 0 128 128"
                    >
                      <path
                        d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                        fill="currentColor"
                      />
                      <path
                        d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </div>
                <div
                  class="css-1vakbk4"
                >
                  <div
                    class="css-k008qs"
                  >
                    <svg
                      class="chakra-icon css-1uwb1nc"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <g
                        stroke="currentColor"
                        stroke-width="1.5"
                      >
                        <path
                          d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                          fill="none"
                          stroke-linecap="round"
                        />
                        <path
                          d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                          fill="currentColor"
                          stroke-linecap="round"
                        />
                        <circle
                          cx="12"
                          cy="12"
                          fill="none"
                          r="11.25"
                          stroke-miterlimit="10"
                        />
                      </g>
                    </svg>
                    <p
                      class="chakra-text css-1qt1967"
                    >
                      20 comments
                    </p>
                  </div>
                  <div
                    class="css-k008qs"
                  >
                    <svg
                      class="chakra-icon css-1uwb1nc"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <g
                        stroke="currentColor"
                        stroke-width="1.5"
                      >
                        <path
                          d="M9,9a3,3,0,1,1,4,2.829,1.5,1.5,0,0,0-1,1.415V14.25"
                          fill="none"
                          stroke-linecap="round"
                        />
                        <path
                          d="M12,17.25a.375.375,0,1,0,.375.375A.375.375,0,0,0,12,17.25h0"
                          fill="currentColor"
                          stroke-linecap="round"
                        />
                        <circle
                          cx="12"
                          cy="12"
                          fill="none"
                          r="11.25"
                          stroke-miterlimit="10"
                        />
                      </g>
                    </svg>
                    <p
                      class="chakra-text css-1qt1967"
                    >
                      15 files
                    </p>
                  </div>
                </div>
              </div>
              <p
                class="chakra-text css-p0439y"
              >
                Sample description for the task
              </p>
            </div>
          </section>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <div
        aria-label="Notifications-top"
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-label="Notifications-top-left"
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-top-right"
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-bottom-left"
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <div
        aria-label="Notifications-bottom"
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <div
        aria-label="Notifications-bottom-right"
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
